lib/c-api/src/wasm_c_api/types/value.rs

```diff
@@ -83,10 +84,11 @@ pub extern "C" fn wasm_valtype_new(kind: wasm_valkind_t) -> Option<Box<wasm_valt
 pub unsafe extern "C" fn wasm_valtype_delete(_valtype: Option<Box<wasm_valtype_t>>) {}
 
 #[no_mangle]
-pub unsafe extern "C" fn wasm_valtype_kind(valtype: *const wasm_valtype_t) -> wasm_valkind_t {
-    if valtype.is_null() {
-        // TODO: handle error
-        panic!("wasm_valtype_kind: argument is null pointer");
-    }
-    return (*valtype).valkind as wasm_valkind_t;
+pub unsafe extern "C" fn wasm_valtype_kind(
+    valtype: Option<NonNull<wasm_valtype_t>>,
```

Comment:
Could we just use `Option<&wasm_valtype_t>` here?

Reply:
We can yes. To be honest, I'm still not sure what the policy regarding `null` value in our implementation or in the Wasm C API.